app/controllers/application_controller/tags.rb

```diff
@@ -86,6 +86,15 @@ def tagging_tags_set_form_vars
     @edit[:current] = copy_hash(@edit[:new])
   end
 
+  def previous_page_url
+    if params[:id]
+      show_url = "/#{params[:controller]}/show"
+      previous_breadcrumb_url == show_url ? "#{show_url}/#{params[:id]}" : previous_breadcrumb_url
+    else
+      params[:miq_grid_checks] ? "/#{params[:controller]}/show_list" : previous_breadcrumb_url
+    end
+  end
```

Comment:
Embedding a ternary in the else is confusing...Think this might be clearer if a little more verbose:

```suggestion
  def previous_page_url
    if params[:id]
      show_url = "/#{params[:controller]}/show"
      previous_breadcrumb_url == show_url ? "#{show_url}/#{params[:id]}" : previous_breadcrumb_url
    else if params[:miq_grid_checks]
      "/#{params[:controller]}/show_list"
    else
      previous_breadcrumb_url
    end
  end
```